ddht/v5_1/network.py

```diff
@@ -449,25 +450,35 @@ async def _manage_routing_table(self) -> None:
                     nursery.start_soon(self._bond, enr.node_id, endpoint)
 
     async def _pong_when_pinged(self) -> None:
-        async with self.dispatcher.subscribe(PingMessage) as subscription:
-            async for request in subscription:
-                await self.dispatcher.send_message(
-                    request.to_response(
-                        PongMessage(
-                            request.request_id,
-                            self.enr_manager.enr.sequence_number,
-                            request.sender_endpoint.ip_address,
-                            request.sender_endpoint.port,
-                        )
-                    )
-                )
+        async def _maybe_add_to_routing_table(
+            request: InboundMessage[PingMessage],
+        ) -> None:
+            try:
                 enr = await self.lookup_enr(
                     request.sender_node_id,
                     enr_seq=request.message.enr_seq,
                     endpoint=request.sender_endpoint,
                 )
-                self.routing_table.update(enr.node_id)
-                self._routing_table_ready.set()
+            except EmptyFindNodesResponse:
```

Comment:
most failed lookups here are timeouts due to being blocked? any other identified reasons?

Reply:
`lookup_enr` is a simple wrapper around doing `FindNodes(0)`.  The `0` distance here has special meaning in the protocol meaning, give me your ENR record.  Mis-behaving/broken/malicious/whatever nodes can return an empty response and this is the exception that will be raised in that case since in all reasonable situations a node can always return their own ENR record.